plugins/woocommerce-blocks/assets/js/blocks/product-collection/types.ts

```diff
@@ -9,6 +9,16 @@ import { type AttributeMetadata } from '@woocommerce/types';
  */
 import { WooCommerceBlockLocation } from '../product-template/utils';
 
+export enum ProductCollectionUIStatesInEditor {
```

Comment:
I like the suggested names, I will update them accordingly. 🙌🏻

> I'm also on the fence on whether the collection chooser state belongs here 🤔

The goal of `ProductCollectionUIStatesInEditor` is to define all possible UI states of Product Collection in Editor. As "Collection Chooser" is a UI state, I think it makes sense to include it here. 

I introduced it because there were two states before: "Collection Chooser" and "Valid." But in this PR, I am introducing another state: "Product Reference Picker." In the future, we need even more states like "Invalid," "Deleted Product Reference," etc. So, the idea is to decide UI state of Product Collection in the Editor at one place to improve maintainability. 

But I am open to suggestions. What do you think? 🤔

Reply:
Sure! I was mainly thinking of this as a "reference-specific" manager, but I don't have a strong opinion, we can handle this as a general state manager 👌